server/src/test/java/org/opensearch/action/support/nodes/TransportQueryGroupStatsActionTests.java

```diff
@@ -0,0 +1,112 @@
+        Map<String, List<MockNodeQueryGroupStatsRequest>> combinedSentRequest = performQueryGroupStatsAction(request);
+
+        assertNotNull(combinedSentRequest);
+        combinedSentRequest.forEach((node, capturedRequestList) -> {
+            assertNotNull(capturedRequestList);
+            capturedRequestList.forEach(sentRequest -> { assertNull(sentRequest.getDiscoveryNodes()); });
+        });
+    }
+
+    private Map<String, List<MockNodeQueryGroupStatsRequest>> performQueryGroupStatsAction(QueryGroupStatsRequest request) {
+        TransportNodesAction action = getTestTransportQueryGroupStatsAction();
+        PlainActionFuture<QueryGroupStatsRequest> listener = new PlainActionFuture<>();
+        action.new AsyncAction(null, request, listener).start();
+        Map<String, List<CapturingTransport.CapturedRequest>> capturedRequests = transport.getCapturedRequestsByTargetNodeAndClear();
+        Map<String, List<MockNodeQueryGroupStatsRequest>> combinedSentRequest = new HashMap<>();
+
+        capturedRequests.forEach((node, capturedRequestList) -> {
+            List<MockNodeQueryGroupStatsRequest> sentRequestList = new ArrayList<>();
+
+            capturedRequestList.forEach(preSentRequest -> {
+                BytesStreamOutput out = new BytesStreamOutput();
+                try {
+                    TransportQueryGroupStatsAction.NodeQueryGroupStatsRequest QueryGroupStatsRequestFromCoordinator =
+                        (TransportQueryGroupStatsAction.NodeQueryGroupStatsRequest) preSentRequest.request;
+                    QueryGroupStatsRequestFromCoordinator.writeTo(out);
+                    StreamInput in = out.bytes().streamInput();
+                    MockNodeQueryGroupStatsRequest QueryGroupStatsRequest = new MockNodeQueryGroupStatsRequest(in);
+                    sentRequestList.add(QueryGroupStatsRequest);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            combinedSentRequest.put(node, sentRequestList);
+        });
+
+        return combinedSentRequest;
+    }
+
+    private TestTransportQueryGroupStatsAction getTestTransportQueryGroupStatsAction() {
+        return new TestTransportQueryGroupStatsAction(
+            THREAD_POOL,
+            clusterService,
+            transportService,
+            mock(QueryGroupService.class),
+            new ActionFilters(Collections.emptySet())
+        );
+    }
+
+    private static class TestTransportQueryGroupStatsAction extends TransportQueryGroupStatsAction {
+        public TestTransportQueryGroupStatsAction(
+            ThreadPool threadPool,
+            ClusterService clusterService,
+            TransportService transportService,
+            QueryGroupService queryGroupService,
+            ActionFilters actionFilters
+        ) {
+            super(threadPool, clusterService, transportService, queryGroupService, actionFilters);
+        }
+    }
```

Comment:
I don't see a reason to have this class ? What is that this class do which original one doesn't